src/System.Device.Gpio/System/Device/I2c/Devices/Windows10I2cDevice.cs

```diff
@@ -77,6 +77,26 @@ public override void Read(Span<byte> buffer)
             new Span<byte>(byteArray).CopyTo(buffer);
         }
 
+        /// <summary>
+        /// Attempts to read data from the I2C device. If the read fails for any reason, returns false rather than throwing an exception.
+        /// </summary>
+        /// <param name="buffer">
+        /// The buffer to read the data from the I2C device.
+        /// The length of the buffer determines how much data to read from the I2C device.</param>
+        /// <returns>true: read succeeded, false if the read failed for any reason.</returns>
+        public override bool ReadWithResult(Span<byte> buffer)
```

Comment:
This should match https://github.com/nanoframework/System.Device.I2c/blob/develop/System.Device.I2c/I2cTransferResult.cs as closely as possible

Reply:
Yep - started implementing that but then realised I'm not sure we have access to the enough level of detail to set those properties. I realise now that pushing to origin was probably a bad idea at this stage as it triggers a review process plus automated tests - nice to know the system works though! 
Am currently cloning the repo on my Linux PC so I can remotely debug with hardware attached, then planning to implement I2cTransferResult with as much info as can be captured from the low level calls